arangod/Agency/Node.cpp

```diff
@@ -82,11 +82,14 @@ void Node::rebuildVecBuf() const {
     Builder tmp;
     {
       VPackArrayBuilder t(&tmp);
-      for (auto const& i : _value) {
-        tmp.add(Slice(i.data()));
+      if (_value != nullptr) {
+        for (auto const& i : *_value) {
+          tmp.add(Slice(i.data()));
+        }
       }
     }
-    _vecBuf = *tmp.steal();
+    _vecBuf = std::make_unique<SmallBuffer>(tmp.size());
+    memcpy(_vecBuf->data(), tmp.data(), tmp.size());
```

Comment:
We have a few such places that create a SmallBuffer object and then memcpy into it.
I would prefer it we added a SmallBuffer constructor that would do this automatically, i.e.
```
SmallBuffer(uint8_t const* data, size_t size) : SmallBuffer() {
  if (size > 0) {
    _start = new uint8_t[size];
    _size = size;
    memcpy(_start, data, size);
  }
}
```
Or maybe a `char const*` instead of `uint8_t const*` in the signature, or both.
Anyway, IMHO all the copying should be done by the SmallBuffer object and not by callers.

Reply:
Good point...